### PR TITLE
remove unused legacy mixer store code

### PR DESCRIPTION
**Pull request by jmuk**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: mixer/pkg/server/server.go [hunk 1]
```diff
@@ -253,3 +253,13 @@ func (s *Server) Close() error {
 func (s *Server) Addr() net.Addr {
 	return s.listener.Addr()
 }
+
+// InventoryMap converts adapter inventory to a builder map.
```
Comment (geeknoid) [inline mixer/pkg/server/server.go]: Is there a better place for this? This function doesn't fit the overall functionality that is offered by the server package.

**Comment by guptasu** [comment]
I think we had a bug earlier. The server.go code should use the function `config.AdapterInfoMap` instead of `InventoryMap`. AdapterInfoMap does validation as well and make sure there are no name collisions. I think we should just delete the `InventoryMap` function from `adapterInfoRegistry.go` and may be leave everything as is. WDYT ?

**Comment by geeknoid** [comment]
For example, seems like this function could go into the adapter package.
This package exposes the Inventory() function already, so it could also
expose InventoryMap...

**Comment by jmuk** [comment]
Hmm, okay, so:
- switched to use `AdapterInfoMap`. I still think it's better to do validation like this on codegen phase, but that's another topic.
- finding the place for adapterInfoRegistry.go is a bit hard, it can't be in `mixer/pkg/adapter` due to some cycle dependency on test. That would be another topic, maybe it would be `mixer/pkg/adapter_registry`, but let's leave it as-is for this specific PR.

Updated the PR. Please take another look.

**Pull request merged**